### DISCUSSION TITLE
Computing power of dipole in lossy/dispersive materials using LDOS

### DISCUSSION THREAD

**Discussion by smartalecH** [NanoComp/meep#2488] [Q&A] [answered]
The [LDOS meep tutorials](https://meep.readthedocs.io/en/latest/Python_Tutorials/Local_Density_of_States/#extraction-efficiency-of-a-light-emitting-diode-led) explain that one can use the LDOS quantities to compute the total emitted power of a dipole:

$$
-\mathrm{Re}(\vec{E} \cdot \vec{J}^*)
$$

As stated in the tutorials, this is rather convenient when the dipole resides near material boundaries and it's impractical to surround the dipole with Poynting flux monitors.

This assumes, however, that the dipole resides in a simple dielectric. The approach described in the tutorials won't work if the material has any conductivities (e.g. is lossy) or polarizabilities.

There *should* be an easy way to account for this, though. For example, a conductivity term in Maxwell's equations can be recast as a current induced by the fields:

$$
\vec{J}_\sigma = \sigma\vec{E}
$$

similarly, the polarizability can be expressed as a current induced by the fields:

$$
\vec{J}_P = \frac{\partial\vec{P}}{\partial t}\cdot\vec{E}
$$

so in theory, we should be able to compute the total power of the dipole using something like:

$$
-\mathrm{Re}(\vec{E} \cdot (\vec{J} + \vec{J}_\sigma + \vec{J}_P)^*)
$$

except I tried this, and it doesn't seem to work...

Am I missing something obvious?

**Comment by stevengj** [comment, marked as answer]
The LDOS isn't well-defined inside a lossy material, because in that case the power expended by a point-dipole antenna is infinite.

You need something else, e.g. the radiative LDOS (which can't be computed from J⋅E, unfortunately; you need the outgoing Poynting flux AFAIK).  Depends on the application, obviously.

**Reply by smartalecH** [reply]
Got it, thanks @stevengj !
